src/ankh/ankh.go

```diff
@@ -107,7 +145,7 @@ func ParseAnkhFile(filename string) (AnkhFile, error) {
 		return config, err
 	}
 
-	err = yaml.Unmarshal(ankhYaml, &config)
+	err = yaml.UnmarshalStrict(ankhYaml, &config)
```

Comment:
I made all the unmarshalling strict so that extra keys don't make it in the yaml files. This should help with typos.

Reply:
i'm torn on this. on one hand we do want to catch errors really early, and on the other hand i'm thinking about how much more difficult it makes upgrading.

we're going to need to exercise discipline when it comes to introducing new fields to the ankh file format (and deprecating old fields). is it possible to mark certain fields as optional? i think it is.

Reply:
I did a little investigation into this issue. Seems like it will only be a problem if we hard remove a field from a struct. We could optionally leave the field in the struct and throw a warning if we want it to be a non-breaking change.

## Initial case

    var data = `
    a: one
    b: two
    `
    type T struct {
      A string
      B string
    }

Result:

    (main.T) {
     A: (string) (len=3) "one",
     B: (string) (len=3) "two"
    }

## Struct field added

    var data = `
    a: one
    b: two
    `
    type T struct {
      A string
      B string
      C string
    }

Result:

    (main.T) {
     A: (string) (len=3) "one",
     B: (string) (len=3) "two",
     C: (string) ""
    }

## User adds field

    var data = `
    a: one
    b: two
    c: three
    `
    type T struct {
      A string
      B string
      C string
    }

Result:

    (main.T) {
     A: (string) (len=3) "one",
     B: (string) (len=3) "two",
     C: (string) (len=5) "three"
    }

## Field is hard removed from struct

    var data = `
    a: one
    b: two
    c: three
    `
    type T struct {
      A string
      B string
    }

Result:

    2018/02/02 16:13:06 error: yaml: unmarshal errors:
      line 4: field c not found in struct main.T
    exit status 1